homeassistant/components/uptimerobot/switch.py

```diff
@@ -0,0 +1,76 @@
+"""UptimeRobot switch platform."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyuptimerobot import UptimeRobotAuthenticationException
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import UptimeRobotDataUpdateCoordinator
+from .const import API_ATTR_OK, DOMAIN, LOGGER
+from .entity import UptimeRobotEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the UptimeRobot switches."""
+    coordinator: UptimeRobotDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [
+            UptimeRobotSwitch(
+                coordinator,
+                SwitchEntityDescription(
+                    key=str(monitor.id),
+                    name=f"{monitor.friendly_name} Active",
+                    device_class=SwitchDeviceClass.SWITCH,
+                ),
+                monitor=monitor,
+            )
+            for monitor in coordinator.data
+        ],
+    )
+
+
+class UptimeRobotSwitch(UptimeRobotEntity, SwitchEntity):
+    """Representation of a UptimeRobot switch."""
+
+    _attr_icon = "mdi:cog"
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if the entity is on."""
+        return bool(self.monitor.status == 2)
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.monitor.status not in (8, 9)
+
+    async def _async_edit_monitor(self, **kwargs: Any) -> None:
+        """Edit monitor status."""
+        try:
+            response = await self.api.async_edit_monitor(**kwargs)
+            self.async_write_ha_state()
+        except UptimeRobotAuthenticationException as exception:
+            raise ConfigEntryAuthFailed(exception) from exception
```

Comment:
This method is called by the async_turn_* methods, nothing is checking for ConfigEntryAuthFailed in those?

Reply:
Check the async_start_reauth method of the ConfigEntry class

Reply:
Done, thx for the hint